code/game/objects/structures/morgue.dm

```diff
@@ -242,67 +247,65 @@ GLOBAL_LIST_EMPTY(crematoriums)
 		else
 			src.icon_state = "crema1"
 
-		if(locked)
+		if(cremating)
 			src.icon_state = "crema_active"
 
 	return
 
-/obj/structure/bodycontainer/crematorium/proc/cremate(mob/user)
-	if(locked)
-		return //don't let you cremate something twice or w/e
-	// Make sure we don't delete the actual morgue and its tray
-	var/list/conts = GetAllContents() - src - connected
+/obj/structure/bodycontainer/crematorium/process()
+	if(cremating)
+		cremate()

```

Comment:
you might as well handle cremation in this proc as well, it's a bit of a buzzword but proc overhead is still an issue.